main/WebServer.cpp

```diff
@@ -12964,7 +12986,8 @@ namespace http
 
 			if (!devoptions.empty())
 			{
-				m_sql.safe_query("UPDATE DeviceStatus SET Options='%q' WHERE (ID == '%q')", devoptions.c_str(), idx.c_str());
+				uint64_t ullidx = std::stoull(idx);
+				m_sql.SetDeviceOptions(ullidx, m_sql.BuildDeviceOptions(devoptions.c_str(), false));
```

Comment:
please remove the .c_str() the function takes a normal std::string

Reply:
Ok done.